analysisservice/src/main/java/nu/yona/server/analysis/service/ActivityPayload.java

```diff
@@ -40,53 +44,30 @@ boolean isNetworkActivity()
 	static ActivityPayload copyTillEndTime(ActivityPayload payload, ZonedDateTime endTime)
 	{
 		return new ActivityPayload(payload.userAnonymized, payload.deviceAnonymized, payload.url, payload.startTime, endTime,
-				payload.application);
+				payload.application, payload.activityCategories);
 	}
 
 	static ActivityPayload copyFromStartTime(ActivityPayload payload, ZonedDateTime startTime)
 	{
 		return new ActivityPayload(payload.userAnonymized, payload.deviceAnonymized, payload.url, startTime, payload.endTime,
-				payload.application);
+				payload.application, payload.activityCategories);
 	}
 
 	static ActivityPayload createInstance(UserAnonymizedDto userAnonymized, DeviceAnonymizedDto deviceAnonymized,
-			NetworkActivityDto networkActivity)
+			NetworkActivityDto networkActivity, Set<ActivityCategoryDto> activityCategories)
 	{
 		ZonedDateTime startTime = networkActivity.getEventTime().orElse(ZonedDateTime.now())
 				.withZoneSameInstant(userAnonymized.getTimeZone());
 		return new ActivityPayload(userAnonymized, deviceAnonymized, Optional.of(networkActivity.getUrl()), startTime, startTime,
-				Optional.empty());
+				Optional.empty(), activityCategories);
 	}
 
 	static ActivityPayload createInstance(UserAnonymizedDto userAnonymized, DeviceAnonymizedDto deviceAnonymized,
-			ZonedDateTime startTime, ZonedDateTime endTime, String application)
+			ZonedDateTime startTime, ZonedDateTime endTime, String application, Set<ActivityCategoryDto> activityCategories)
 	{
 		ZoneId userTimeZone = userAnonymized.getTimeZone();
 		return new ActivityPayload(userAnonymized, deviceAnonymized, Optional.empty(),
-				startTime.withZoneSameInstant(userTimeZone), endTime.withZoneSameInstant(userTimeZone), Optional.of(application));
-	}
-
-	@Override
-	public boolean equals(Object other)
```

Comment:
Any idea why we had ``equals`` and ``hashCode`` before and now don't need it anymore?

Reply:
It was introduced by me in PR #459 but used for unit tests only. I thought it was better to revert it now.

Reply:
OK, if the tests can work without it then it's indeed better to remove it.